File: Correct/1학년/C. 체스는 어려워/main.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

int main() {
    unordered_map<char, int> score = {
        {'Q', 9}, {'R', 5}, {'B', 3}, {'N', 3}, {'P', 1},
        {'q', 9}, {'r', 5}, {'b', 3}, {'n', 3}, {'p', 1}
    };

    int white = 0, black = 0;
    string row;

    for (int i = 0; i < 8; ++i) {
        cin >> row;
        for (char ch : row) {
            if (ch == '.') continue;
            if (score.find(ch) == score.end()) continue;
            if (isupper(ch))
                black += score[ch];
            else
                white += score[ch];
        }
    }

    if (white > black)
        cout << "White" << endl;
    else if (black > white)
        cout << "Black" << endl;
    else
        cout << "Draw" << endl;

    return 0;
}

File: Correct/1학년/C. 체스는 어려워/main.py
# 말 점수 정의
score_table = {
    'q': 9, 'r': 5, 'b': 3, 'n': 3, 'p': 1,
    'k': 0
}

white_score = 0
black_score = 0

for _ in range(8):
    line = input()
    for ch in line:
        if ch == '.':
            continue
        ch_lower = ch.lower()
        if ch_lower not in score_table:
            continue
        if ch.isupper():
            black_score += score_table[ch_lower]
        else:
            white_score += score_table[ch_lower]

if white_score > black_score:
    print("White")
elif black_score > white_score:
    print("Black")
else:
    print("Draw")